concrete/src/Session/SessionFactory.php

```diff
@@ -285,38 +283,70 @@ protected function getRedisHandler(array $config)
      *  Decides whether to return a Redis Instance or RedisArray Instance depending on the number of servers passed to it.
      *
      * @param array $servers The `concrete.session.servers` or `concrete.session.redis.servers` config item
+     * @param int | null $database The concrete.session.redis.database config item
      *
-     * @return \Redis | \RedisArray
+     * @return \Redis | \RedisArray | \Predis\Client
      */
-    private function getRedisInstance(array $servers)
+    private function getRedisInstance(array $servers, $database = null)
     {
+        $database = !empty($database)?(int) $database : 1;
+        // This is a way to load our predis client without creating new instances
+        if ($this->app->isShared(Client::class)) {
+            $redis = $this->app->make(Client::class);
```

Comment:
Turns out the bind from `DatabaseServiceProvider.php` breaks `this->app->make()` so this works around it

But I'll change either this to use a singleton name, or change that